### PR TITLE
Mobt 157 nbhood refactor tidy clis

### PR DESCRIPTION
**Pull request by fionaRust**
Patially Addresses https://github.com/metoppv/mo-blue-team/issues/157
Removes unnecessary options from CLIs and ensure full acceptance test coverage for the new usage

New acceptance test data location on the ticket.

Changed acceptance test data plots:
| Test | Old or square version | New | Diff if not obvious | Notes |
| -- | -- | -- | --  |-- |
|New test collapse band circular test in nbhood_iterate_with_mask |kgo_collapsed_square| kgo_collaped_circular | kgo_collapsed_diff | differences between the original square version of this test and the new circular version are small |
 | New kgo for test_masked_square_external in test_nbhood  |  kgo_external_masked_old    |kgo_external_masked_new   |  |Now reapply the mask by default.  |
| New kgo for test_masked_square_remask in test_nbhood | kgo_masked_old   |   kgo_masked_new  |  |Now reapply the mask by default.  | 
| new land sea mask test for nbhood_land_and_sea   | no_top_bands_square  | no_top_bands_circular | no_top_bands_diff  | differences between the original square version of this test and the new circular version are small | 
| new topographic bands test for nbhood_land_and_sea | top_bands_square | top_bands_circular| top_bands_diff | differences between the original square version of this test and the new circular version are small |

 

Testing:
 - [ ] Ran tests and they passed OK
 - [ ] Added new tests for the new feature(s)

CLA
 - [ ] If a new developer, signed up to CLA

### CODE REVIEW

File: improver_tests/acceptance/test_nbhood_land_and_sea.py [hunk 1]
```diff
@@ -48,7 +48,37 @@ def test_basic(tmp_path):
...
+        "--radii",
+        "20000",
+        "--output",
+        output_path,
+    ]
+    run_cli(args)
+    acc.compare(output_path, kgo_path)
+
+
+def test_basic_circular(tmp_path):
```
Comment (bayliffe) [inline improver_tests/acceptance/test_nbhood_land_and_sea.py:65]: Same again, if not writing it in the doc-string, the test above needs a `_square` suffix.

File: improver_tests/acceptance/test_nbhood_land_and_sea.py [hunk 2]
```diff
@@ -174,6 +230,32 @@ def test_topographic_bands_probabilities(tmp_path):
...
+        "20000",
+        "--output",
+        output_path,
+    ]
+    run_cli(args)
+    acc.compare(output_path, kgo_path)
+
+
+@pytest.mark.slow
+def test_topographic_bands_probabilities_circular(tmp_path):
```
Comment (bayliffe) [inline improver_tests/acceptance/test_nbhood_land_and_sea.py:245]: Here as well.

**Review by bayliffe — CHANGES_REQUESTED** [review]
A few comments, nothing major.

File: improver/cli/nbhood_iterate_with_mask.py [hunk 3]
```diff
@@ -76,6 +77,11 @@ def process(
         coord_for_masking (str):
             String matching the name of the coordinate that will be used
             for masking.
+        neighbourhood_shape (str):
+            Name of the neighbourhood method to use. Only a "circular"
+            neighbourhood shape is applicable for calculating "percentiles"
+            output.
```
[the remark this answers is not in the document]
Comment (fionaRust) [reply improver/cli/nbhood_iterate_with_mask.py:83]: Sorry too much copy and pasting.

File: improver/cli/nbhood_land_and_sea.py [hunk 4]
```diff
@@ -68,6 +69,11 @@ def process(
             A cube containing the weights which are used for collapsing the
             dimension gained through masking. These weights must have been
             created using a land-sea mask. (Optional).
+        neighbourhood_shape (str):
+            Name of the neighbourhood method to use. Only a "circular"
+            neighbourhood shape is applicable for calculating "percentiles"
+            output.
```
[the remark this answers is not in the document]
Comment (fionaRust) [reply improver/cli/nbhood_land_and_sea.py:75]: And again

File: improver/cli/nbhood_iterate_with_mask.py [hunk 5]
```diff
@@ -88,12 +94,7 @@ def process(
             lead_times. Lead times must be given as integer values.
         area_sum (bool):
             Return sum rather than fraction over the neighbourhood area.
-        remask (bool):
```
[the remark this answers is not in the document]
Comment (fionaRust) [reply improver/cli/nbhood_iterate_with_mask.py:91]: Removed

File: improver/nbhood/use_nbhood.py [hunk 6]
```diff
@@ -172,27 +175,16 @@ def __init__(
                 If False, use a circle with constant weighting.
             sum_only:
                 If true, return neighbourhood sum instead of mean.
-            re_mask:
```
[the remark this answers is not in the document]
Comment (fionaRust) [reply improver/nbhood/use_nbhood.py:175]: Changed

File: improver_tests/acceptance/test_nbhood_iterate_with_mask.py [hunk 7]
```diff
@@ -78,6 +80,34 @@ def test_collapse_bands(tmp_path):
...
+        "10000",
+        "--output",
+        output_path,
+    ]
+    run_cli(args)
+    acc.compare(output_path, kgo_path)
+
+
+@pytest.mark.slow
+def test_collapse_bands_circular(tmp_path):
```
[the remark this answers is not in the document]
Comment (fionaRust) [reply improver_tests/acceptance/test_nbhood_iterate_with_mask.py:95]: I think I've changed all of these now

**Review by bayliffe — APPROVED** [review]

**Review by gavinevans — CHANGES_REQUESTED** [review]
Thanks @fionaRust 👍 

@mspelman07 and I have added a few comments.

File: improver/cli/nbhood_iterate_with_mask.py [hunk 8]
```diff
@@ -43,6 +43,7 @@ def process(
     weights: cli.inputcube = None,
     *,
     coord_for_masking,
+    neighbourhood_shape,
```
[the remark this answers is not in the document]
Comment (fionaRust) [reply improver/cli/nbhood_iterate_with_mask.py:46]: Yep I missed this, now removed

File: improver_tests/acceptance/test_nbhood.py [hunk 9]
```diff
@@ -168,7 +168,6 @@ def test_masked_square_remask(tmp_path):
         "square",
         "--radii",
         "20000",
-        "--remask",
```
[the remark this answers is not in the document]
Comment (fionaRust) [reply improver_tests/acceptance/test_nbhood.py:171]: Yes it is, I've removed it and the corresponding KGO.

File: improver_tests/acceptance/test_nbhood_iterate_with_mask.py [hunk 10]
```diff
@@ -78,6 +80,34 @@ def test_collapse_bands(tmp_path):
...
+        "--radii",
+        "10000",
+        "--output",
+        output_path,
+    ]
+    run_cli(args)
+    acc.compare(output_path, kgo_path)
+
+
+@pytest.mark.slow
```
[the remark this answers is not in the document]
Comment (fionaRust) [reply improver_tests/acceptance/test_nbhood_iterate_with_mask.py:94]: Done

File: improver_tests/acceptance/test_nbhood_land_and_sea.py [hunk 11]
```diff
@@ -41,14 +41,44 @@
 run_cli = acc.run_cli(CLI)
 
 
-def test_basic(tmp_path):
+def test_basic_square(tmp_path):
```
[the remark this answers is not in the document]
Comment (fionaRust) [reply improver_tests/acceptance/test_nbhood_land_and_sea.py:44]: Done

File: improver_tests/acceptance/test_nbhood_land_and_sea.py [hunk 12]
```diff
@@ -156,13 +203,22 @@ def test_landsea_only(tmp_path, landsea):
...
+        "--output",
+        output_path,
+    ]
     run_cli(args)
     acc.compare(output_path, kgo_path)
 
 
 @pytest.mark.slow
-def test_topographic_bands_probabilities(tmp_path):
+def test_topographic_bands_probabilities_square(tmp_path):
```
[the remark this answers is not in the document]
Comment (fionaRust) [reply improver_tests/acceptance/test_nbhood_land_and_sea.py:221]: Done

File: improver_tests/acceptance/test_nbhood_iterate_with_mask.py [hunk 13]
```diff
@@ -91,27 +95,27 @@ def test_collapse_bands_square(tmp_path):
...
-        "--neighbourhood-shape",
-        "circular",
-        "--radii",
-        "10000",
-        "--output",
-        output_path,
-    ]
-    run_cli(args)
-    acc.compare(output_path, kgo_path)
+# @pytest.mark.slow
```
Comment (gavinevans) [inline improver_tests/acceptance/test_nbhood_iterate_with_mask.py:98]: I think that this can now be removed.

**Review by gavinevans — CHANGES_REQUESTED** [review]
Thanks @fionaRust 👍 

This looks good apart from some commented out code in the `test_nbhood_iterate_with_mask.py` file that should be removed.

File: improver_tests/acceptance/test_nbhood_iterate_with_mask.py (hunk 13 above)
Comment (fionaRust) [reply improver_tests/acceptance/test_nbhood_iterate_with_mask.py]: Done

**Review by gavinevans — APPROVED** [review]

**Pull request merged**